lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -170,33 +170,47 @@ public static function get_theme_data() {
 	 *
 	 * It can also create and return a new draft CPT.
 	 *
-	 * @param bool  $should_create_cpt Whether a new CPT should be created if no one was found.
-	 *                                 False by default.
-	 * @param array $post_status_filter Filter CPT by post status.
-	 *                                  ['publish'] by default, so it only fetches published posts.
+	 * @param WP_Theme $theme             The theme object.
+	 *                                    If empty, it defaults to the current theme.
+	 * @param bool     $should_create_cpt Whether a new CPT should be created if no one was found.
+	 *                                    False by default.
+	 * @param array    $post_status_filter Filter CPT by post status.
+	 *                                    ['publish'] by default, so it only fetches published posts.
 	 *
 	 * @return array Custom Post Type for the user's origin config.
 	 */
-	private static function get_user_data_from_custom_post_type( $should_create_cpt = false, $post_status_filter = array( 'publish' ) ) {
+	public static function get_user_data_from_custom_post_type( $theme, $should_create_cpt = false, $post_status_filter = array( 'publish' ) ) {
```

Comment:
Hi, @xknown

What's the reason for adding the `$theme` argument?

I understand visibility change is required for unit tests, but can we use `::get_user_data()` like your initial test in the issue?

Reply:
The visibility/signature change is required to remove the duplicate code block from `gutenberg_add_active_global_styles_link`.

Reply:
Sorry, I just noticed that part.